Build failure: protoc-gen-js

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build nixpkgs/946738808f0ddc9e57ee0df76aa5986715c24676#protoc-gen-js`
or
`NIXPKGS_ALLOW_UNFREE=1 nix build nixpkgs/946738808f0ddc9e57ee0df76aa5986715c24676#anytype --impure`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317515273

### Relevant log output

```console
ERROR: /build/output/external/com_google_absl/absl/flags/BUILD.bazel:175:11: Compiling absl/flags/reflection.cc failed: (Exit 1): gcc failed: error executing CppCompile command (from target @@com_google_absl//absl/flags:reflection) /nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++14' ... (remaining 35 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/cassert:46,
                 from external/com_google_absl/absl/container/internal/container_memory.h:18,
                 from external/com_google_absl/absl/container/flat_hash_map.h:40,
                 from external/com_google_absl/absl/flags/reflection.h:29,
                 from external/com_google_absl/absl/flags/reflection.cc:16:
external/com_google_absl/absl/container/internal/container_memory.h: In function 'void* absl::lts_20240116::container_internal::Allocate(Alloc*, size_t)':
external/com_google_absl/absl/container/internal/container_memory.h:66:27: error: 'uintptr_t' does not name a type [-Wtemplate-body]
   66 |   assert(reinterpret_cast<uintptr_t>(p) % Alignment == 0 &&
      |                           ^~~~~~~~~
external/com_google_absl/absl/container/internal/container_memory.h:31:1: note: 'uintptr_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
   30 | #include "absl/utility/utility.h"
  +++ |+#include <cstdint>
   31 |
```

### Additional context

Likely https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.1-zen1, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/gep/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/gep/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/gep/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/gep/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/sa7kxynphvibmp8z8dar31j3dyn3nvfj-lix-2.93.3/share`
 - nixpkgs: `/nix/store/1ny7brxnqbx6xilj7mjdlinzpb5a1s3i-source`

### Notify maintainers

No maintainers for `protoc-gen-js`

cc anytype maintainers: @autrimpo @Adda0 @kira-bruneau 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pr/protoc-gen-js/package.nix
+++ b/pkgs/by-name/pr/protoc-gen-js/package.nix
@@ -1,5 +1,6 @@
 {
   stdenv,
+  gcc14Stdenv,
   lib,
   buildBazelPackage,
   bazel_7,
@@ -7,7 +8,17 @@
   cctools,
 }:
 
-buildBazelPackage rec {
+let
+  # fails to build with gcc15, see https://github.com/NixOS/nixpkgs/issues/475586
+  buildBazelPackage' =
+    if stdenv.cc.isGNU then
+      buildBazelPackage.override {
+        stdenv = gcc14Stdenv;
+      }
+    else
+      buildBazelPackage;
+in
+buildBazelPackage' rec {
   pname = "protoc-gen-js";
   version = "3.21.4";
 

